.studio/01_base.sh

```diff
@@ -0,0 +1,288 @@
+
+    [[ "${3:-}" != "silent" ]] && log_line "Waiting to load svc '$(yellow "$1")'. ($(yellow "$COUNTER of $SECONDS_WAITING"))";
+  done
+}
+
+document "wait_or_fail_for_port_to_listen" <<DOC
+  Wait for a port to be listening. If the port is not found in the set amount of time this helper returns 1.
+  @(arg:1) Port to wait for to be listening
+  @(arg:2) Number of seconds to wait before returning 1. (default: 60 seconds)
+  @(arg:3) Wheter or not this process runs silently
+  Example: Wait for a service to start listening on port 1234 for 1 minute (defaults)
+  ---------------------------------------------
+  wait_or_fail_for_port_to_listen 1234
+  Example: Wait for a service to start listening on port 1234 for 10 seconds
+  ---------------------------------------------
+  wait_or_fail_for_port_to_listen 1234 10
+DOC
+wait_or_fail_for_port_to_listen() {
+  install_if_missing core/busybox-static netstat
+  local SECONDS_WAITING=${2:-60}
+
+  local COUNTER=0
+  while ! netstat -an | grep "$1" | grep LISTEN &>/dev/null; do
+    sleep 1
+
+    if [[ $COUNTER -ge "$SECONDS_WAITING" ]]; then
+      error "Failed listening for port '$1'"
+      return 1
+    fi
+
+    (( COUNTER=COUNTER+1 ))
+    [[ "${3:-}" != "silent" ]] && log_line "Waiting for port '$(yellow "$1")' to be listening ($(yellow "$COUNTER of $SECONDS_WAITING"))";
+  done
+}
+
+document "wait_for_ok_response" <<DOC
+  Wait a set amount of time for the provided URL to response with status code (200).
+  If the time expires then this helper returns 1
+  @(arg:1) URL to curl
+  @(arg:2) Number of seconds to wait before returning 1. (default: 60 seconds)
+  Example: Wait for www.google.com to return Ok for 1 minute
+  ---------------------------------------------
+  wait_or_fail_for_ok_response www.google.com 60
+  Example: Wait for www.google.com to return Ok for 10 seconds
+  ---------------------------------------------
+  wait_or_fail_for_ok_response www.google.com 10
+DOC
+wait_for_ok_response() {
+  local url="${1:?Missing url argument; try 'describe wait_for_ok_response'}"
+  local SECONDS_TO_WAIT=${2:-60}
+
+  code=200;
+  install_if_missing core/curl curl
+  local COUNTER=0
+  local response
+
+  while [[ $COUNTER -le "$SECONDS_TO_WAIT"  ]]; do
+    # shellcheck disable=SC1083
+    response="$(curl --write-out %{http_code} --silent --output /dev/null "$url")"
+    if [[ $response -eq $code ]]; then
+      break
+    else
+      log_line "Waiting for '$url' to response OK ($code). [Got:$response] ($(yellow "$COUNTER of $SECONDS_TO_WAIT"))"
+      sleep 1
+      (( COUNTER=COUNTER+1 ))
+    fi
+
+    if [[ $COUNTER -ge "$SECONDS_WAITING" ]]; then
+      return 1
+    fi
+  done
+}
+
+document "install_go_tool" <<DOC
+  Install the specified go tool(s).
+  @(arg:*) The array of packages you wish to install.
+  The default behavior is to install go tools with 'go install -v -mod=vendor'
+
+  To install tools using not included in the vendor you'll either need to
```

Comment:
Not quite sure of the intent, maybe...
```suggestion
  To install tools not included in the vendor directory, you'll either need to
```